README-USERS.md

```diff
@@ -77,31 +77,123 @@ TODO: AWS provides its own wrapper around Docker through ECS. We will need to ab
+>>> assert '^docker/' in response_text
+
+# On that route, requests are proxied to containers by name:
+>>> proxy_url = django_url + '/docker/' + container_name + '/'
+>>> proxy_url
+'http://localhost:8000/docker/basic-nginx/'
+>>> assert 'Welcome to nginx' in requests.get(proxy_url).text
+
 ```
-Note the URL that's returned: You'll get the Nginx welcome page if you visit it.
-You can run `docker ps` to see the container you've started.
 
-Now let's see how proxying works:
+### Please wait
+
+By default, if the container is not yet responding to
+requests, the proxy will return a "Please wait" page with both a 
+JS- and a meta-reload. You can customize the content of this page, 
+and the reload behavior.
+
+Note, though, that there is no attempt to distinguish between a container
+that is taking its time, and one that may never start up: In either case,
+the user just gets the "Please wait" page by default, and it will continue
+to refresh indefinitely.
 
 ```
-$ python manage.py runserver &
-$ curl http://localhost:8000/docker/my-server/ | grep title
-<title>Welcome to nginx!</title>
+# Make sure Django is still up:
+>>> response_text = requests.get(django_url).text
+>>> assert '^docker/' in response_text
+
+# Try to get a container that doesn't exist:
+>>> container_name = 'please-wait'
+>>> proxy_url = django_url + '/docker/' + container_name + '/'
+>>> proxy_url
+'http://localhost:8000/docker/please-wait/'
+>>> response_text = requests.get(proxy_url).text
+>>> assert 'Please wait' in response_text
+>>> assert 'http-equiv="refresh"' in response_text
+
+```
+
+### Historian
+
+The `Historian` provides access to a record of the requests that have been
+made through the Proxy. Its configuration, along with that for the "Please wait"
+page needs to be made as part of your django configuration. The example here
+shows how it can be configured:
+
 ```
-Django receives the request, looks up the container from the name in the URL path,
-proxies the request, and returns the same Nginx welcome page.
\ No newline at end of file
+>>> from os import environ
+>>> environ['DJANGO_SETTINGS_MODULE'] = 'demo_path_routing.settings'
+>>> from django_docker_engine.historian import FileHistorian
+>>> from django_docker_engine.proxy import Proxy
+>>> from django.test import RequestFactory
+
+>>> import django
+>>> django.setup()
+
+>>> historian = FileHistorian('/tmp/readme-doc-text.txt')
+>>> title_text = 'test-title'
+>>> body_html = '<p>test-body</p>'
+>>> proxy = Proxy(
+...     historian=historian,
+...     please_wait_title='<test-title>',
+...     please_wait_body_html='<p>test-body</p>')
+>>> urlpatterns = proxy.url_patterns()
+
+# All of this would be handled by Django in practice:
+>>> response = urlpatterns[0].callback(
+...     request=RequestFactory().get('/fake-url'),
+...     container_name='fake-container',
+...     url='fake-url')
+>>> html = response.content.decode()
+>>> assert '<title>&lt;test-title&gt;</title>' in html
```

Comment:
I think we're more or less making the same assertions on the Please Wait example above, maybe just the final assert to illustrate the `historian` functionality?

Reply:
This is actually cool, dont change